### PR TITLE
Feature/edit enroll on activity details

### PR DESCRIPTION
**Pull request by hmaryam84**
### This PR implements the correct button in Activity Details
 * When the user is on Activity Details screen, in this case either:

   - The activity is in active mode:
    -He is logged in and he is not the creator of the activity : in this case an enroll button will show up
    -He is in guest mode so a log in/ Register button shows up
    -He is the creator of the activity and an edit button shows up

   - The activity is in finished mode:
    - a text indicating that the activity is not active is displayed
  
* Concerning the tests, they check if each type of button is correctly displayed in all different situations and that it navigates to the correct screen when needed.

### CODE REVIEW

**Comment by elvan03** [comment]
The code looks great, nice touch to put Toasts when enrolling is successful or fails.
LGTM!

File: app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt [hunk 1]
```diff
@@ -183,28 +182,54 @@ fun ActivityDetailsScreen(
...
-                        listActivityViewModel.updateActivity(theActivity)
-                        profileViewModel.addActivity(profile.id, theActivity.uid)
-                        Toast.makeText(context, "Enroll Successful", Toast.LENGTH_SHORT).show()
-                        navigationActions.navigateTo(Screen.OVERVIEW)
-                      } else {
-                        Toast.makeText(
-                                context,
-                                "Enroll failed, limit of places reached",
-                                Toast.LENGTH_SHORT)
-                            .show()
```
Comment (DAFERMOH) [inline app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt:200]: good thinking about showing the enroll button only if it is a user and not the creator of the activity.

File: app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt [hunk 2]
```diff
@@ -183,28 +182,54 @@ fun ActivityDetailsScreen(
...
+                              .show()
+                        }
+                      },
+                      modifier =
+                          Modifier.fillMaxWidth()
+                              .padding(horizontal = 24.dp)
+                              .testTag("enrollButton")) {
+                        Text(text = "Enroll")
+                      }
+                } else {
```
Comment (DAFERMOH) [inline app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt:210]: good that the creator of the activity can navigate to the editActivityScreen to edit his activity.

File: app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt [hunk 3]
```diff
@@ -183,28 +182,54 @@ fun ActivityDetailsScreen(
...
+                  Button(
+                      onClick = { navigationActions.navigateTo(Screen.EDIT_ACTIVITY) },
+                      modifier =
+                          Modifier.fillMaxWidth()
+                              .padding(horizontal = 24.dp)
+                              .testTag("editButton")) {
+                        Text(text = "Edit")
                       }
-                    },
+                }
```
Comment (DAFERMOH) [inline app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt:219]: LGTM!

**Review by DAFERMOH — APPROVED** [review]
LGTM!

**Pull request merged**